task_processing/plugins/mesos/timeout_executor.py

```diff
@@ -0,0 +1,110 @@
+import logging
+import time
+from threading import Lock
+from threading import Thread
+
+from cllist import dllist
+from six.moves.queue import Queue
+
+from task_processing.interfaces.task_executor import TaskExecutor
+
+log = logging.getLogger(__name__)
+
+
+class TimeoutExecutor(TaskExecutor):
+    def __init__(self, downstream_executor, timeout_s):
+        self.downstream_executor = downstream_executor
+        if timeout_s <= 0:
+            raise ValueError('TimeoutExecutor expects a positive timeout')
+        else:
+            self.timeout_s = timeout_s
```

Comment:
I wonder if we should make the timeout a per task thing, rather than something that exists on the executor level? I think that might give more flexibility